common/rule-processor/src/RuleHelper.js

```diff
@@ -107,8 +111,9 @@ class RuleMatcher {
         return null;
```

Comment:
Shouldn't this also return object?? Upstream method might break due to `.destination` here

Reply:
```
if(result) {
    result?.destination
}
else {
    return null
}
```